src/Equinox.Cosmos/Cosmos.fs

```diff
@@ -536,12 +539,22 @@ module Sync =
     let batch (log : ILogger) retryPolicy containerStream batch: Async<Result> =
         let call = logged containerStream batch
         Log.withLoggedRetries retryPolicy "writeAttempt" call log
+
     let mkBatch (stream: string) (events: IEventData<_>[]) unfolds: Tip =
         {   p = stream; id = Tip.WellKnownDocumentId; n = -1L(*Server-managed*); i = -1L(*Server-managed*); _etag = null
             e = [| for e in events -> { t = e.Timestamp; c = e.EventType; d = e.Data; m = e.Meta; correlationId = e.CorrelationId; causationId = e.CausationId } |]
             u = Array.ofSeq unfolds }
-    let mkUnfold baseIndex (unfolds: IEventData<_> seq) : Unfold seq =
-        unfolds |> Seq.mapi (fun offset x -> { i = baseIndex + int64 offset; c = x.EventType; d = x.Data; m = x.Meta; t = DateTimeOffset.UtcNow } : Unfold)
+
+    let mkUnfold compress baseIndex (unfolds: IEventData<_> seq) : Unfold seq =
+        unfolds
+        |> Seq.mapi (fun offset x ->
+            {
+                i = baseIndex + int64 offset
+                c = x.EventType
+                d = x.Data |> if compress then JsonCompressedBase64Converter.Compress else id
```

Comment:
can assign function to temp on L549 though I guess splitting it out is not outright bad!

Reply:
Yea I was thinking of doing it but then decided to avoid the extra allocation. But I can definitely switch it to that.

Reply:
are you sure this all collapses in the IL to no allocations though ? (I'd have inlined the pipe and removed the DRY to avoid the question, but really the chances are I'd have made a temp in order to continue to shoehord it onto one line ;) )
It's all good, just remarking

Reply:
No worries, I updated it to use a temp function